src/spec_coerce/core.cljc

```diff
@@ -389,12 +449,23 @@
   "Recursively coerce map values on a structure."
   ([x] (coerce-structure x {}))
   ([x {::keys [overrides op]
-       :or    {op coerce}}]
+       :or    {op coerce}
+       :as    opts}]
    (walk/prewalk (fn [x]
-                   (if (map? x)
-                     (with-meta (into {} (map (fn [[k v]]
-                                                (let [coercion (get overrides k k)]
-                                                  [k (op coercion v)]))) x)
-                       (meta x))
-                     x))
-     x)))
+                   (cond->> x
+                     (map? x)
+                     (into (empty x)
+                           (map (fn [[k v]]
+                                  (let [coercion (get overrides k k)]
+                                    [k (op coercion v opts)]))))))
+                 x)))
+
+(s/def ::foo (s/nilable string?))
```

Comment:
Can we move these examples as tests? Or maybe just remove then from here

Reply:
done, I already had tests covering these. I just removed these examples